TestProject/gameoflife.php

```diff
@@ -112,6 +117,23 @@ function myAutoload($className) {
 else die ("Could not find input $requestedInput!\n");
 
 
+//Initialize and return the defined classes inside the output directory
+$requestedOutput = $options->getOption("output") ?? "ConsoleOutput";
+$classNameForOutput = "output\\$requestedOutput";
+
+if(class_exists($classNameForOutput))
+{
+    $output = new $classNameForOutput;
+    if($output instanceof BaseOutput)
+    {
+        $output->outputBoard($board);
+        $output->finishOutput();
```

Comment:
Here you should only call the method `startOutput()` so that the output can (if necessary) prepare itself.
The call to `$output->outputBoard($board)` must be moved down into the loop and should replace the existing line `$board->printOutBoard()`.
finsishOutput() should then be called once after the loop.